### PR TITLE
Reset outdated isFrontPage before saving new front page

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14254
affects: Student
release note: Fixed an issue with a cached course home page

### CODE REVIEW

File: Core/Core/Pages/GetPage.swift [hunk 1]
```diff
@@ -36,6 +36,18 @@ public struct GetPage: UseCase {
...
+    public func reset(context: NSManagedObjectContext) {
+        guard isFrontPage else { return }
+        // Unset previous flags to ensure we get the new front page
+        let frontPages: [Page] = context.fetch(NSCompoundPredicate(andPredicateWithSubpredicates: [
+            NSPredicate(key: #keyPath(Page.contextID), equals: self.context.canvasContextID),
+            NSPredicate(format: "%K == true", #keyPath(Page.isFrontPage)),
+        ]))
+        for front in frontPages {
+            front.isFrontPage = false
+        }
```
Comment (bluwave) [inline Core/Core/Pages/GetPage.swift:48]: do you need to call save on the context?

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/n91an0a) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fn91an0a)
teacher: [Link](https://rebrand.ly/nw7yf0n) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fnw7yf0n)
parent: [Link](https://rebrand.ly/jswhyf7) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fjswhyf7)

File: Core/Core/Pages/GetPage.swift (hunk 1 above)
Comment (vanwagonet) [reply Core/Core/Pages/GetPage.swift:48]: no, the usecase boilerplate calls save after `reset` and `write`.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Fixed an issue with a cached course home page

#### Affected Apps: Student

[MBL-14254](https://instructure.atlassian.net/browse/MBL-14254)

<details><summary>:x: XCTest failed: CoreTests/GradeListViewControllerTests/testLayout</summary>


    Assertion Failure: GradeListViewControllerTests.swift:136: XCTAssertEqual failed: ("Optional("One")") is not equal to ("Optional("Two")")
    Assertion Failure: GradeListViewControllerTests.swift:140: XCTAssertEqual failed: ("Optional("Complex Numbers")") is not equal to ("Optional("Proof that proofs are useful")")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**